File: OCPlayground/OCPlayground/MyClass.m
//
//  MyClass.m
//  OCPlayground
//
//  Created by zj－db0465 on 15/8/19.
//  Copyright (c) 2015年 icetime17. All rights reserved.
//

#import "MyClass.h"

@implementation MyClass

- (id)init {
    self = [super init];
    if (self) {
        
    }
    return self;
}

- (id)initWithName:(NSString *)name age:(NSInteger)age {
    self = [super init];
    if (self) {
        _name = name;
        _age = age;
    }
    return self;
}

- (void)desc {
    NSLog(@"%@ : %ld", self.name, (long)self.age);
}

- (void)descPerformSelector {
    NSLog(@"%@ : %ld", self.name, (long)self.age);
}

@end

File: DemoSwift/DemoSwift/DemoSwift/MyClass.swift
//
//  MyClass.swift
//  DemoSwift
//
//  Created by zj－db0465 on 15/11/17.
//  Copyright © 2015年 icetime17. All rights reserved.
//

import UIKit

typealias MyClosure = (String, Int) -> String

class MyClass: NSObject {

    var name = "My Class"
    var age = 10
    
    private var privateName = "Private Name"
    
    internal var internalName = "Internal Name"
    
    var myClosure = MyClosure?()
    
    func myFunc() {
        print("My Func")
    }
    
    func myPrivate() {
        print(self.privateName)
    }
    
    func myClosureFunc(closure: MyClosure) -> String {
        return closure(name, age)
    }
    
    
    var mySchool: String {
        get {
            return "SJTU"
        }
        
        set {
            print("mySchool set method")
        }
    }
    
}
